packages/parse-push/src/main.ts

```diff
@@ -0,0 +1,82 @@
+import * as core from '@actions/core'
+import * as gh from '@actions/github'
+import * as Webhooks from '@octokit/webhooks'
+import * as fs from 'fs'
+import * as path from 'path'
+
+// /**
+//  * Extract the list of exercises from a list of files.
+//  *
+//  * @param files contains a list of files
+//  */
+// async function getExercises(files: string[]): Promise<Set<string>> {
+//     const re = /exercises\/([a-z-]*)\//
+//     let exs = new Set<string>()
+
+//     for (const f of files) {
+//         const m = re.exec(f)
+//         if (m) {
+//             exs.add(m[0])
+//         }
+//     }
+
+//     return exs
+// }
+
+async function run(): Promise<void> {
+    try {
+        // Confirm that it's a push event
+        if (gh.context.eventName != 'push') {
+            throw new Error(`Event ${gh.context.eventName} is not supported`)
+        }
+
+        const payload = gh.context.payload as Webhooks.EventPayloads.WebhookPayloadPush
+
+        // To prevent mistakes, only act on pushes to main or master branch
+        // TODO: Remove probby-tests from this list
+        if (
+            !(
+                payload.ref == 'refs/heads/main' ||
+                payload.ref == 'refs/heads/master' ||
+                payload.ref == 'refs/heads/probby-tests'
+            )
+        ) {
+            throw new Error(
+                `Only pushes to the default branch should trigger notifications. Perhaps you have misconfigured the workflow?`,
+            )
+        }
+
+        // TODO: Process commits
+        // PR URL can be guessed if the commit message ends with (#1746) which will work for most squashed PRs
+        // We _could_ parse recently merged PRs but that seems like too much hassle to be worth it
```

Comment:
What's also an option is to actually commit it to and retrieve it from a new .toml file. That way, it can also be manually fixed.

Reply:
Turns out there's an API endpoint to find associated PRs: https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-pull-requests-associated-with-a-commit